src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -211,4 +226,39 @@ private void validateMedicalInventory(MedicalInventory medInventory) throws OHDa
 			throw new OHDataValidationException(errors);
 		}
 	}
+	
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void validateInventory(MedicalInventory inventory, List<MedicalInventoryRow> inventoryRowSearchList) throws OHServiceException {
+		LocalDateTime movFrom = inventory.getLastModifiedDate();
+		LocalDateTime movTo = TimeTools.getNow();
+		List<Movement> movements = movBrowserManager.getMovements(null, null, null, null, movFrom, movTo, null, null, null, null); 
+		if (!movements.isEmpty()) {
+			Map<Medical, List<Movement>> groupedByMedical = movements.stream().collect(Collectors.groupingBy(Movement::getMedical));
+			for (Iterator<MedicalInventoryRow> iterator = inventoryRowSearchList.iterator(); iterator.hasNext();) {
+				MedicalInventoryRow medicalInventoryRow = (MedicalInventoryRow) iterator.next();
+				String lotCode = medicalInventoryRow.getLot().getCode();
+				double theoQty = medicalInventoryRow.getTheoreticQty();
+				Medical medical = medicalInventoryRow.getMedical();
+				List<Movement> movs = groupedByMedical.get(medical);
+				if (movs != null) {
+					Movement mov = movs.get(0);
+					String lotCodeOfMovement = mov.getLot().getCode();
+					Lot lot = movStockInsertingManager.getLot(lotCodeOfMovement);
+					double mainStoreQty = (double)lot.getMainStoreQuantity();
+					Integer medicalCode = medical.getCode();
+					Integer movMedicalCode = mov.getMedical().getCode();
+					if (movMedicalCode.equals(medicalCode) && lotCodeOfMovement.equals(lotCode)) {
+						if (mainStoreQty != theoQty) { 
+							medicalInventoryRow.setTheoreticQty(mainStoreQty);
+							medicalInventoryRow.setRealqty(mainStoreQty);
+							medicalInventoryRowManager.updateMedicalInventoryRow(medicalInventoryRow);
```

Comment:
Maybe I've not posted this comment: if a theoretic quantity is updated, it should also be notified to the user, in order to bring his attention on verify the real quantity (that in turn, should NOT be updated by the validation as well). So I remark again that the validate method should return a list of "errors" (i.e. messages) to be displayed to the user like other validation methods do.